### PR TITLE
C#: Re-generate .NET 8 Runtime models.

### PR DESCRIPTION
**Pull request by michaelnebel**
Re-generating the .NET runtime models revealed the very broad set of database (reader) sources.

Since we are now generating models that applies to all subclasses of database reader, the issue with result multiplicity gets a lot bigger, if we still perceive all database reader expressions as sources. As a result, we need to adjust the database reader sources. Combined with the new summaries, it reduces result multiplicities and yields better path explanations. Furthermore, it turned out that we needed to promote some of the summaries from generated to manual as there are some cases where flow is not identified for the overrides in source code, which causes us to loose results for the `mono` repository.

### CODE REVIEW

**Comment by michaelnebel** [comment]
For the first DCA execution (remote flow sources) there are 5 new results. 1 in the `powershell` project and 4 in the `mono` project. I spot checked the result in the `powershell` project and this seems sound. The result path uses at least the following new summary:
```
"System.Reflection", "Assembly", True, "get_Location", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"
```
The path consists around 100 steps, but it appears that the assembly location is "stored" which appears to be a true positive.

**Comment by michaelnebel** [comment]
The "lost" `sql-injection` results from local sources are because flow is not identified via the source code overrides of `GetString` (`IDataRecord` member). To fix this issue we promote the generated summaries to `manual` as they are then still considered even though the source code is present.

**Comment by michaelnebel** [comment]
The latest DCA run looks good; I have also spot checked one of the local source results.
The new `cs/sql-injection` result on `mono` is because of the summary 
```
["System.Data", "IDataRecord", True, "GetString", "(System.Int32)", "", "Argument[this]", "ReturnValue", "taint", "manual"]
```
which looks sound.

That is, with the update .NET 8 Runtime models we 
- Get 15 new results on our DCA projects.
- A reduction in the multiplicity of database sourced results AND better (longer) path explanations for database sources.

File: csharp/ql/lib/ext/System.Data.model.yml [hunk 1]
```diff
@@ -54,6 +54,17 @@ extensions:
       - ["System.Data", "EnumerableRowCollectionExtensions", False, "Where<TRow>", "(System.Data.EnumerableRowCollection<TRow>,System.Func<TRow,System.Boolean>)", "", "Argument[0].Element", "ReturnValue.Element", "value", "manual"]
       - ["System.Data", "IColumnMappingCollection", True, "get_Item", "(System.String)", "", "Argument[this].Element", "ReturnValue", "value", "manual"]
       - ["System.Data", "IColumnMappingCollection", True, "set_Item", "(System.String,System.Object)", "", "Argument[1]", "Argument[this].Element", "value", "manual"]
+      - ["System.Data", "IDataRecord", True, "GetDataTypeName", "(System.Int32)", "", "Argument[this]", "ReturnValue", "taint", "manual"]
```
Comment (tamasvajk) [inline csharp/ql/lib/ext/System.Data.model.yml:57]: This API seems to return a value from a fixed set of datatypes. Do we need this model?

File: csharp/ql/lib/ext/System.Data.model.yml [hunk 2]
```diff
@@ -54,6 +54,17 @@ extensions:
...
+      - ["System.Data", "IDataRecord", True, "GetDateTime", "(System.Int32)", "", "Argument[this]", "ReturnValue", "taint", "manual"]
```
Comment (tamasvajk) [inline csharp/ql/lib/ext/System.Data.model.yml:58]: This API returns a `DateTime`, do we need to consider this API in taint tracking?

File: csharp/ql/lib/ext/System.Data.model.yml [hunk 3]
```diff
@@ -54,6 +54,17 @@ extensions:
...
+      - ["System.Data", "IDataRecord", True, "GetFieldType", "(System.Int32)", "", "Argument[this]", "ReturnValue", "taint", "manual"]
```
Comment (tamasvajk) [inline csharp/ql/lib/ext/System.Data.model.yml:59]: This API returns a `System.Type`. Do we need this model?

File: csharp/ql/lib/ext/System.Data.model.yml (hunk 1 above)
Comment (michaelnebel) [reply csharp/ql/lib/ext/System.Data.model.yml:57]: No, I will introduce a neutral for this one and see if we can avoid generating it alltogether!

File: csharp/ql/lib/ext/System.Data.model.yml (hunk 2 above)
Comment (michaelnebel) [reply csharp/ql/lib/ext/System.Data.model.yml:58]: Generally `DateTime` has started to be considered a more "simple" type (a form of santizer) for most queries, so we should definitely exclude such summaries from the model generation in general! I will modify the generator and re-generate (thank you for pointing this out!!!).

File: csharp/ql/lib/ext/System.Data.model.yml (hunk 3 above)
Comment (michaelnebel) [reply csharp/ql/lib/ext/System.Data.model.yml:59]: No, we probably shouldn't. It appears that java already exclude generating models like this. I will do the same for C#!
Thank you :-)

Comment (michaelnebel) [reply csharp/ql/lib/ext/System.Data.model.yml:59]: We should probably also introduce neutrals for `GetGuid` and `GetName` to avoid false positives.

**Comment by michaelnebel** [comment]
DCA still looks good.

**Review by tamasvajk — APPROVED** [review]

**Pull request merged**